internal/provider/resource_auth0_user.go

```diff
@@ -259,49 +259,60 @@ func deleteUser(ctx context.Context, d *schema.ResourceData, m interface{}) diag
 func expandUser(d *schema.ResourceData) (*management.User, error) {
 	config := d.GetRawConfig()
 
-	user := &management.User{
-		Connection: value.String(config.GetAttr("connection_name")),
-		Name:       value.String(config.GetAttr("name")),
-		GivenName:  value.String(config.GetAttr("given_name")),
-		FamilyName: value.String(config.GetAttr("family_name")),
-		Nickname:   value.String(config.GetAttr("nickname")),
-		Picture:    value.String(config.GetAttr("picture")),
-		Blocked:    value.Bool(config.GetAttr("blocked")),
-	}
+	user := &management.User{}
 
 	if d.IsNewResource() {
 		user.ID = value.String(config.GetAttr("user_id"))
 	}
-	if d.IsNewResource() || d.HasChange("email") {
```

Comment:
The `d.IsNewResource` is unnecessary as the d.HasChange will also trigger when this is a new resource needing to be created.